drivers/ethernet/eth_sam_gmac.c

```diff
@@ -1785,16 +1785,22 @@ static void get_mac_addr_from_i2c_eeprom(uint8_t mac_addr[6])
 {
 	const struct device *dev;
 	uint32_t iaddr = CONFIG_ETH_SAM_GMAC_MAC_I2C_INT_ADDRESS;
+	int ret;
 
 	dev = device_get_binding(CONFIG_ETH_SAM_GMAC_MAC_I2C_DEV_NAME);
 	if (!dev) {
 		LOG_ERR("I2C: Device not found");
 		return;
 	}
 
-	i2c_write_read(dev, CONFIG_ETH_SAM_GMAC_MAC_I2C_SLAVE_ADDRESS,
-		       &iaddr, CONFIG_ETH_SAM_GMAC_MAC_I2C_INT_ADDRESS_SIZE,
-		       mac_addr, 6);
+	ret = i2c_write_read(dev, CONFIG_ETH_SAM_GMAC_MAC_I2C_SLAVE_ADDRESS,
+			   &iaddr, CONFIG_ETH_SAM_GMAC_MAC_I2C_INT_ADDRESS_SIZE,
+			   mac_addr, 6);
+
+	if (ret != 0) {
+		LOG_ERR("I2C: i2c_write_read() failed");
```

Comment:
I wanted to give this comment with previous PR (regarding samples/*), but there it's not so important. But this driver can be used in production code, and if log an error at, let's make it be helpful for user/engineer to understand what's the problem. Something like "failed to read MAC addr" would ok (and that's optimized for code size, "failed to read MAC addr from EEPROM" would be better, but too long).

Reply:
@pfalcon I agree. I made the change. Let me know if you'd like to drop the `"I2C: "` part.